app/api/users.py

```diff
@@ -214,6 +214,14 @@ def before_update_object(self, user, data, view_kwargs):
         if has_access('is_admin') and data.get('deleted_at') != user.deleted_at:
             user.deleted_at = data.get('deleted_at')
 
+        if has_access('is_user_itself', user_id=user.id) and data.get('deleted_at') != user.deleted_at:
+            if len(user.events) != 0:
```

Comment:
This is wrong. User associated with active events can't be deleted. Same with orders

Reply:
@iamareebjamal when we discussed this earlier, we were trying to fathom the scams which an individual can do. We also concluded these possibilities -
 - User can create an event and do not show up on the venue and deletes his/her account just after that. This should not be the case 
 - User buys free tickets for an event and does not show up. Similarly deletes his/her account just after the event.

As we were discussing this we concluded to restrict this functionality only for users who have no events or orders associated as a first step. If we change the condition and only restrict users with active events/orders to delete their account, how can we avoid above menti scams?

Reply:
An event cannot be deleted if it has active orders - and hence a user can't delete its account. I'm approving this. But need to evaluate and ensure that an order can never be deleted by a user

Reply:
Currently, we are not allowing any user to delete his/her orders. @iamareebjamal